examples/generative/corrdiff/train.py

```diff
@@ -90,6 +90,9 @@ def main(cfg: DictConfig) -> None:
     c.out_channels = getattr(cfg, "out_channels", [0, 17, 18, 19])
     c.img_shape_x = getattr(cfg, "img_shape_x", 448)
     c.img_shape_y = getattr(cfg, "img_shape_y", 448)
+    c.patch_shape_x = getattr(cfg, "patch_shape_x", 448)
+    c.patch_shape_y = getattr(cfg, "patch_shape_y", 448)
+    c.patch_num = getattr(cfg, "patch_num", 1)
```

Comment:
Could `patch_num` be inferred from `path_shape` instead of making it configurable?

Reply:
It's quite challenging. The selection of patch_num is highly dependent on GPU memory, patch shape, batch size, and other factors to optimize memory usage effectively. Determining the optimal value might require extensive testing across various scenarios. Therefore, I think setting it as a tunable parameter might be a good idea?

Reply:
yeah sounds good.